### PR TITLE
Centralize clippy lints in workspace

### PR DESCRIPTION
**Pull request by rmalmain**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -86,6 +87,62 @@ which = "6.0.3"
...
+missing_panics_doc = "allow"
+missing_docs_in_private_items = "allow"
+module_name_repetitions = "allow"
+ptr_cast_constness = "allow"
+negative_feature_names = "allow"
+unsafe_derive_deserialize = "allow"
+similar_names = "allow"
+too_many_lines = "allow"
+into_iter_without_iter = "allow"        # Broken
+borrow_as_ptr = "allow"                 # Till they fix this buggy lint in clippy
```
Comment (domenukk) [inline Cargo.toml:131]: Can we check if this still applys?

File: Cargo.toml [hunk 2]
```diff
@@ -86,6 +87,62 @@ which = "6.0.3"
...
+type_repetition_in_bounds = "allow"
+missing_errors_doc = "allow"
+cast_possible_truncation = "allow"
+used_underscore_binding = "allow"
+ptr_as_ptr = "allow"
+missing_panics_doc = "allow"
+missing_docs_in_private_items = "allow"
+module_name_repetitions = "allow"
+ptr_cast_constness = "allow"
+negative_feature_names = "allow"
```
Comment (domenukk) [inline Cargo.toml:126]: Let's not put this one in the main clippy allow

File: Cargo.toml (hunk 1 above)
Comment (rmalmain) [reply Cargo.toml:131]: was gonna check, for now i just wanted to copy paste and unify before tweaking stuff around

File: fuzzers/binary_only/qemu_launcher/src/harness.rs [hunk 3]
```diff
@@ -98,7 +98,7 @@ impl Harness {
         }
         let len = len as GuestReg;
 
-        unsafe { self.qemu.write_mem(self.input_addr, buf) };
+        self.qemu.write_mem(self.input_addr, buf).unwrap();
```
Comment (domenukk) [inline fuzzers/binary_only/qemu_launcher/src/harness.rs:101]: `.map_err(|e| Error::unknown(format!("Failed to write memory to {:?}: {e:?}", self.input_addr)))?;`

But anyway qemu doesn't seem to return an error?

Comment (domenukk) [reply fuzzers/binary_only/qemu_launcher/src/harness.rs:101]: Ah you changed it. Nice!

**Pull request merged**